delfin/api/schemas/alert_source.py

```diff
@@ -14,23 +14,24 @@
 
 from delfin.api.validation import parameter_types
 
-
+# engineId is in range (5-32) octet which is 10-64 hex characters
+# If it is odd length, 0 will be prefixed to last octet, so minimum length is 9
 put = {
     'type': 'object',
     'properties': {
         'host': parameter_types.hostname_or_ip_address,
         'version': parameter_types.snmp_version,
         'community_string': {'type': 'string',
                              'minLength': 1,
-                             'maxLength': 255},
-        'username': {'type': 'string', 'minLength': 1, 'maxLength': 255},
+                             'maxLength': 32},
```

Comment:
Are these limits based on any RFC standards? or pysnmp limitation?

Reply:
These are mainly as per pysnmp convention but more or less inline with rfc support
For ex: username, engineid, community str etc are same as rfc
For auth priv keys, it depends on algorithm we use, so they have fixed rage which we followed